.github/workflows/build-wheel-macos-arm64.yaml

```diff
@@ -339,6 +339,7 @@ jobs:
 
     - name: Build wheel
       run: |
+        python${{ matrix.python_version }} -m pip install setuptools
```

Comment:
`sysconfig` is in the standard library no, or does this installation have a different purpose?

Reply:
It has a different purpose. pybind11 will attempt to import distutils if setuptools is not installed. This is necessary for intree_extension that we are importing from pybind11

[See here](https://github.com/pybind/pybind11/blob/master/pybind11/setup_helpers.py#L65-L72)

```python
try:
    from setuptools import Extension as _Extension
    from setuptools.command.build_ext import build_ext as _build_ext
except ImportError:
    from distutils.command.build_ext import (  # type: ignore[assignment]
        build_ext as _build_ext,
    )
    from distutils.extension import Extension as _Extension  # type: ignore[assignment]
```

Reply:
Oh good to know, however I have a few questions:
- We should always try to manage dependencies automatically first, and only do manual installations if necessary. I'm a bit surprised that setuptools isn't available on a system to begin with, but we actually already have it as a package build dependency: https://github.com/PennyLaneAI/catalyst/blob/a3562a485b87ec0adcf9317d14dd1b4c618530d3/pyproject.toml#L31
  Do you have any idea why the build prerequisite in the pyproject.toml is not sufficient?
- Why is this only needed on the arm mac runner?

Either way we can merge this to unblock the release, but we should leave a TODO that this should be removed for the above reasons.

Reply:
> Why is this only needed on the arm mac runner?

This is not only needed on the arm mac runner. I was just testing.

> We should always try to manage dependencies automatically first, and only do manual installations if necessary. I'm a bit surprised that setuptools isn't available on a system to begin with, but we actually already have it as a package build dependency: 

I agree with you, I was surprised as well. I am not the most familiar with the python build ecosystem myself.

Reply:
Thanks, let's go ahead with this temporary fix then :)

Reply:
Just figured out why `setuptools` is not installed when building the wheel, even though it is listed in the pyproject.toml. 

The pyproject.toml build prerequisites are only used if you use `pip`, which we do when installing the package for development, but not when we create the wheel. Here we invoke the setup script directly `python setup.py bdist_wheel`.

Will update.